smartapps/ady624/webcore-piston.src/webcore-piston.groovy

```diff
@@ -801,958 +554,1032 @@ private checkVersion(rtData) {
 def fakeHandler(event) {
 	def rtData = getRunTimeData()
 	warn "Received unexpected event [${event.device?:location}].${event.name} = ${event.value} (device has no active subscriptions)... ", rtData
-    updateLogs(rtData)
+	updateLogs(rtData)
 }
 
 def deviceHandler(event) {
 	handleEvents(event)
 }
 
 def timeHandler(event, recovery = false) {
-try {
-	handleEvents([date: new Date(event.t), device: location, name: 'time', value: event.t, schedule: event, recovery: recovery])
-    } catch (Exception e) {
-    error "Unexpected error:", null, null, e
-    }
+	try {
+		handleEvents([date: new Date(event.t), device: location, name: 'time', value: event.t, schedule: event, recovery: recovery])
+	} catch (Exception e) {
+		error "Unexpected error:", null, null, e
+	}
 }
 
+/*
 //new and improved timeout recovery management
 def timeoutRecoveryHandler_webCoRE(event) {
 	timeHandler([t:now()], true)
 }
+*/
 
 def timeRecoveryHandler(event) {
-	timeHandler(event, true)
+	if(event) {
```

Comment:
Good catch. This has bugged me for awhile since the recovery handler wouldn't actually do anything since it was null. With a local scheduler though, I wonder the need for it unless something went really wrong in the piston's processing, since there is a lot of error catching that happens to hopefully prevent that.